index.js

```diff
@@ -44,23 +44,47 @@ module.exports = function(options){
       promise = getSpecsFromPattern(specGlobs);
     }
     return promise.then(arr => {
-      return _.sortBy(arr, 'name');
+      return _.sortBy(arr, getDisplayOption());
     });
   }
 
+  function getCommonStartPath(files) {
+    files = files.sort();
+    const first = files[0];
+    const last = files[files.length - 1];
+    let rv = '';
+    if (first !== last) {
+      let i;
+      for (i = 0; i < first.length && i < last.length && first.charAt(i) === last.charAt(i); ++i) {
+        // No body, just leading i along
+      }
+      rv = first.substring(0, i);
+    }
+    return rv;
+  }
+
+  function getDisplayOption(spec) {
+    const fileLongEnough = spec ? spec.file.length > spec.name.length : true;
+    return (options.showPath && fileLongEnough) ? 'file' : 'name';
+  }
+
   function getSpecsFromPattern(pattern){
     return new Promise((res, rej) => {
       glob(pattern, go, (err, files) => {
           if (err) {
               rej(err);
           } else {
               res(files.map(file => {
+                const commonStart = getCommonStartPath(files);
```

Comment:
can you move this up 1 line (outside of the files.map call)? No need to call it multiple times.

Reply:
Done. Thanks!